warehouse/age-off-utils/src/test/java/datawave/age/off/util/AgeOffCsvToMatchPatternFormatterTest.java

```diff
@@ -0,0 +1,394 @@
+        // @formatter:off
+        String input = HEADER_WITHOUT_LABEL + "\n" +
+                "bakingPowder, 365d\n" +
+                "driedBeans\n" + // missing duration
+                "      coffeeGround       ,        90d\n          " +
+                "coffeeWholeBean        ,183d\n" +
+                "         coffeeInstant,730d\n";;
+        // @formatter:on
+
+        reformat(new Builder(), input);
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void failsWithMissingHeaderDurationToken() throws IOException {
+        // @formatter:off
+        String input = "bakingPowder, 365d\n";
+        // @formatter:on
+
+        reformat(new Builder(), input);
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void failsWithEmptyDurationToken() throws IOException {
+        // @formatter:off
+        String input = HEADER_WITHOUT_LABEL + "\n" +
+                "bakingPowder, 365d\n" +
+                "driedBeans,\n" + // empty duration
+                "      coffeeGround       ,        90d\n          " +
+                "coffeeWholeBean        ,183d\n" +
+                "         coffeeInstant,730d\n";;
+        // @formatter:on
+
+        Builder builder = new Builder();
+
+        reformat(builder, input);
+    }
+
+    private String prepareInputWithOverride() {
+        // add an override for driedBeans from 548d to 365d
+        String inputWithOneOverride = adjustEachLine(INPUT_TEXT, line -> {
+            if (line.contains("driedBeans")) {
+                return line + ",365d";
+            }
+            return line + ",";
+        });
+
+        return HEADER_WITHOUT_LABEL + ",override" + "\n" + inputWithOneOverride;
+    }
+
+    private String readFileContents(String shelfLifeFileOut) throws IOException, URISyntaxException {
+        List<String> expectedFileContents = Files.readAllLines(new File(this.getClass().getResource(shelfLifeFileOut).toURI()).toPath());
+        return expectedFileContents.stream().collect(Collectors.joining("\n"));
+    }
+
+    static Builder setShelfLifeInputFile(Builder builder) throws IOException, URISyntaxException {
+        return builder.setInputFile(new File(AgeOffCsvToMatchPatternFormatterTest.class.getResource(SHELF_LIFE_FILE_IN).toURI()));
+    }
+
+    private String reformat(Builder builder, String inputText) throws IOException {
+        builder.setReader(new StringReader(inputText));
+        return reformat(builder);
+    }
+
+    private String reformat(Builder builder) throws IOException {
+        StringWriter out = new StringWriter();
+        builder.setWriter(out);
+        AgeOffCsvToMatchPatternFormatter generator = new AgeOffCsvToMatchPatternFormatter(builder.build());
+        generator.write();
+
+        return out.toString();
+    }
+
+    private static String adjustEachLine(String input, Function<String,String> function) {
+        return Arrays.stream(input.split("\\n")).map(item -> {
```

Comment:
The `map()` method actually accepts a function, so you could do

```
Arrays.stream(input.split("\\n")).map(function).collect(Collectors.joining(...)
```
Maybe for readability, you could rename `function` to `lineFormatter` or something like that.

Reply:
Good catch!  Addressed in latest push.